src/Npgsql/Internal/Converters/Networking/MacaddrConverter.cs

```diff
@@ -0,0 +1,36 @@
+using System.Buffers;
+using System.Diagnostics;
+using System.Net.NetworkInformation;
+
+namespace Npgsql.Internal.Converters;
+
+sealed class MacaddrConverter : PgBufferedConverter<PhysicalAddress>
+{
+    public override bool CanConvert(DataFormat format, out BufferingRequirement bufferingRequirement)
+    {
+        bufferingRequirement = BufferingRequirement.FixedSize;
+        return base.CanConvert(format, out _);
+    }
+
+    public override Size GetSize(SizeContext context, PhysicalAddress value, ref object? writeState)
+        => value is null // TODO: Remove this
```

Comment:
Note that null values end up here - ToBufferRequirements passes `null!` which gets converted to `default!` in GetSizeAsObject (so this seems like a bug with buffered converters over reference types).

Reply:
As discussed offline, fixed size conversions shouldn't have to inspect the value. Just remove the CanConvert override to make it do the expected thing.

`PgBufferedConverter` defaults to `BufferingRequirement.Value`
`PgStreamingConverter` defaults to `BufferingRequirement.None`